src/main/java/haste/commands/Parser.java

```diff
@@ -0,0 +1,282 @@
+            break;
+        case "find":
+            c = verifyFind(words);
+            break;
+        case "mark":
+            c = verifyMark(words);
+            break;
+        case "unmark":
+            c = verifyUnmark(words);
+            break;
+        case "delete":
+            c = verifyDelete(words);
+            break;
+        case "todo":
+            c = verifyTodo(cmd);
+            break;
+        case "deadline":
+            c = verifyDeadline(cmd);
+            break;
+        case "event":
+            c = verifyEvent(cmd);
+            break;
+        default:
+            c = new InvalidCommand("Command should start with bye/list/mark/unmark/todo/deadline/event");
+        }
+
+        return c;
+    }
+
+    public static Command verifyFind(String[] words) {
+        if (words.length != 2) {
+            return new InvalidCommand("input (only) 1 word after \"find\"!");
+        }
+        String keyword = words[1];
+        return new FindCommand(keyword);
+    }
+
+    /**
+     * Returns a command that deletes task.
+     * Verifies user intends to delete task.
+     *
+     * @param words The array of words user inputted.
+     * @return DeleteCommand object containing index of task.
+     */
+    public static Command verifyDelete(String[] words) {
+        int id;
+
+        if (words.length != 2) {
+            return new InvalidCommand("input (only) number after \"delete\"!");
+        }
+        try {
+            id = Integer.parseInt(words[1]) - 1;
+        } catch (NumberFormatException e) {
+            return new InvalidCommand("only numerical input allowed!");
+        }
+        if (id >= TaskList.numOfTasks || id < 0) {
+            return new InvalidCommand("task does not exist! input within the list of numbered tasks.");
+        }
+
+        return new DeleteCommand(id);
+    }
+
+    /**
+     * Returns a command that marks task.
+     * Verifies user intends to mark task.
+     *
+     * @param words The array of words user inputted.
+     * @return MarkCommand object containing index of task.
+     */
+    public static Command verifyMark(String[] words) {
+        int id ;
+
+        if (words.length != 2) {
+            return new InvalidCommand("input (only) number after \"mark\"!");
+        }
+        try {
+            id = Integer.parseInt(words[1]) - 1;
+        } catch (NumberFormatException e) {
+            return new InvalidCommand("only numerical input allowed!");
+        }
+        if (id >= TaskList.numOfTasks || id < 0) {
+            return new InvalidCommand("task does not exist! input within the list of numbered tasks.");
+        }
+
+        return new MarkCommand(id);
+    }
+
+    /**
+     * Returns a command that unmarks task.
+     * Verifies user intends to unmark task.
+     *
+     * @param words The array of words user inputted.
+     * @return MarkCommand object containing index of task.
+     */
+    public static Command verifyUnmark(String[] words) {
```

Comment:
Does this do the same thing as verifyMark? If so, I think this method can be omitted.

Reply:
It seems that verifyMark and verifyUnmark are identical with two exceptions - one returns a new MarkCommand while the other returns a UnmarkCommand and a small change in one of the texts.

@yihfei, maybe you could consider using a helper function that both verifyMark and verifyUnmark use?